conformance-tests/src/main.rs

```diff
@@ -0,0 +1,264 @@
+use std::{
+    path::{Path, PathBuf},
+    process::{Command, Stdio},
+};
+
+use anyhow::Context as _;
+use test_environment::{
+    http::{Request, Response},
+    io::OutputStream,
+    services::ServicesConfig,
+    Runtime, TestEnvironment, TestEnvironmentConfig,
+};
+
+fn main() {
+    let tests_dir = conformance_tests::download_tests().unwrap();
+    let mut args = std::env::args().skip(1);
+    let spin_binary = &args
+        .next()
+        .expect("expected first arg to be path to Spin binary");
+    let ctr_binary = &args
+        .next()
+        .expect("expected second arg to be path to ctr binary");
+
+    'test: for test in conformance_tests::tests_iter(&tests_dir).unwrap() {
```

Comment:
Is the reason you're not using `conformance_tests::run_tests` because the tests run in parallel? Should we provide an API that allows the tests to not run in parallel? 

That API is preferred since it gives better test output based on [`libtest-mimic`](https://github.com/LukasKalbertodt/libtest-mimic).

Reply:
I wasn't aware of the `run_tests` API being added. I can change to use that.

Reply:
Running them in parallel we run into issues concurrently pushing to the local registry

Reply:
Okay, got them running in parallel. There was an issue with the registry service not coming up. Commit [c3a0a58](https://github.com/spinkube/containerd-shim-spin/pull/134/commits/c3a0a5821cca833d3570dc5d289d0ce0374f96bb) leverages `run_test` and updates these tests to configure the port for the Spin app and generate a random hash for the run ID to enable running the tests in parallel.